sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/SonarScanner.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Orchestrator
+ * Copyright (C) 2011 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package com.sonar.orchestrator.build;
+
+import com.google.common.collect.ImmutableMap;
+import com.sonar.orchestrator.config.Configuration;
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Executes the sonar-runner script. In-process mode is not supported yet.
+ *
+ * @since 2.1
+ */
+public class SonarScanner extends SonarRunner {
+
+  SonarScanner() {
+  }
+
+  @Override
+  protected Map<String, String> doGetEnvironmentVariablePrefixes() {
+    // http://jira.sonarsource.com/browse/ORCH-256
+    // Temporarily hardcoded in Orchestrator meanwhile sonar-runner 2.5
+    return ImmutableMap.of("SONAR_RUNNER_OPTS", "-Djava.awt.headless=true");
+  }
+
+  @Override
+  public SonarScanner setRunnerVersion(String s) {
+    return (SonarScanner) super.setRunnerVersion(s);
+  }
+
+  @Override
+  public SonarScanner setProjectDir(File dir) {
+    return (SonarScanner) super.setProjectDir(dir);
+  }
+
+  @Override
+  public SonarScanner setProjectKey(@Nullable String s) {
+    return (SonarScanner) super.setProjectKey(s);
+  }
+
+  @Override
+  public SonarScanner setProjectVersion(@Nullable String s) {
+    return (SonarScanner) super.setProjectVersion(s);
+  }
+
+  @Override
+  public SonarScanner setProjectName(@Nullable String s) {
+    return (SonarScanner) super.setProjectName(s);
+  }
+
+  @Override
+  public SonarScanner setSourceDirs(@Nullable String s) {
+    return (SonarScanner) super.setSourceDirs(s);
+  }
+
+  @Override
+  public SonarScanner setTestDirs(@Nullable String testDirs) {
+    return (SonarScanner) super.setTestDirs(testDirs);
+  }
+
+  /**
+   * @deprecated since 3.8 sonar.binaries is deprecated
+   */
+  @Deprecated
+  @Override
+  public SonarScanner setBinaries(@Nullable String s) {
+    return (SonarScanner) super.setBinaries(s);
+  }
+
+  /**
+   * @deprecated since 3.8 sonar.libraries is deprecated
+   */
+  @Deprecated
+  @Override
+  public SonarScanner setLibraries(@Nullable String s) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)